app/src/main/java/com/jakdor/apapp/ui/apartment/ApartmentImageAdapter.kt

```diff
@@ -0,0 +1,60 @@
+package com.jakdor.apapp.ui.apartment
+
+import android.graphics.Bitmap
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.cardview.widget.CardView
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.RequestManager
+import com.bumptech.glide.request.RequestOptions
+import com.jakdor.apapp.R
+import java.util.*
+
+class ApartmentImageAdapter(private val glide: RequestManager, private val imagesList: ArrayList<Picture>):
+    RecyclerView.Adapter<ApartmentImageAdapter.Holder>() {
+
+    lateinit var recyclerViewItemClickListener: RecyclerViewItemClickListener
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
+        val inflater = LayoutInflater.from(parent.context)
+
+        return Holder(inflater.inflate(R.layout.image_apartment_list, parent, false))
+    }
+
+    override fun getItemCount(): Int {
+        return imagesList.size
+    }
+
+    override fun onBindViewHolder(holder: Holder, position: Int) {
+        holder.itemPosition = position
+        val item = imagesList[position].image
+
+        glide.load(item)
+            .apply(
+                RequestOptions()
+                    .fitCenter()
+                    .placeholder(R.mipmap.ic_launcher_round)
+            )
+            .centerCrop()
+            .into(holder.imageView)
+    }
+
+    fun setOnItemClickListener(recyclerViewItemClickListener: RecyclerViewItemClickListener) {
+        this.recyclerViewItemClickListener = recyclerViewItemClickListener
+    }
+
+    inner class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+        var imageView: ImageView = itemView.findViewById(R.id.apartment_image)
+        var cardView: CardView = itemView.findViewById(R.id.image_cardView)
+        var itemPosition: Int = 0
+
+        init {
+            cardView.setOnClickListener { recyclerViewItemClickListener.onItemClick(itemView, itemPosition) }
```

Comment:
Przeniósł bym to do onBindViewHolder, teraz nie ma takiej sytuacji, ale jak dodasz DiffCallback to może być tak, że usuniesz pozycję i coś dodasz i Ci się wszystko rozjedzie